samples/net/sockets/coap_server/src/coap-server.c

```diff
@@ -97,6 +100,8 @@ static bool join_coap_multicast_group(void)
 		return false;
 	}
 
+	net_ipv6_mld_join(iface, &mcast_addr.sin6_addr);
```

Comment:
The `net_ipv6_mld_join()` will check if the maddr is already added to the interface and if it is not, then it will add it. So the call `net_if_ipv6_maddr_add()` above could be omitted. We should check the return value of the join call here and print some error if the joining failed.

Reply:
Ok, thanks, will check it in my project first.